### DISCUSSION TITLE
使用kafka作为消息队列的问题

### DISCUSSION THREAD

**Discussion by caoping6** [milvus-io/milvus#24185] [Q&A and General discussion]
querynode的消费groupid生成方式是 by-dev-rootcoord-dml-0-{collectionId}-{nodeId}

1. 那么一个节点是否会因为多个集合生成多个消费组？
2. 每个消费组消费的都是当前topic的全量数据，是怎么筛选数据的？
3. 如果当前节点挂了，querycoord如何协调，会在其他节点上生成新的消费组吗？之前的消费组如何处理？

**Comment by LoveEachDay** [comment]
1. yes
2. dml channel topic 里的消息都带有collection metadata, 每个consumer group 会按collection过滤，如果多个collection 复用了同一个dml topic
3. yes, 会在别的querynode 会watch dml topic,  之前创建的consumer group 不会清理

**Reply by xiaofan-luan** [reply]
> 1. yes
> 2. dml channel topic 里的消息都带有collection metadata, 每个consumer group 会按collection过滤，如果多个collection 复用了同一个dml topic
> 3. yes, 会在别的querynode 会watch dml topic,  之前创建的consumer group 不会清理

会在别的querynode 会watch dml topic,  之前创建的consumer group 不会清理,这个是符合预期的么？我理解应该用同一个cunsumer group合理点
